helix-view/src/graphics.rs

```diff
@@ -17,6 +19,36 @@ pub enum CursorKind {
     Hidden,
 }
 
+impl Default for CursorKind {
+    fn default() -> Self {
+        Self::Block
+    }
+}
+
+impl FromStr for CursorKind {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "bar" => Ok(Self::Bar),
+            "block" => Ok(Self::Block),
+            "underline" => Ok(Self::Underline),
+            _ => Err(anyhow!("Invalid cursor '{}'", s)),
+        }
+    }
+}
+
+// toml deserializer doesn't seem to recognize string as enum
+impl<'de> Deserialize<'de> for CursorKind {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        s.parse().map_err(de::Error::custom)
+    }
+}

```

Comment:
Do we have to do this ourselves? Can't just use `#[serde(rename = "camelCase")]`?

Reply:
You're right, thanks !